atlasaction/action.go

```diff
@@ -693,6 +699,37 @@ func (a *Actions) SchemaApply(ctx context.Context) error {
 	return nil
 }
 
+// ApprovalChecks checks the approval checks for the schema plan.
+// It will fail the action if the approval checks are not satisfied.
+func (a *Actions) ApprovalChecks(ctx context.Context, tc *TriggerContext) error {
+	if !a.GetBoolInput("require-approval") {
+		return nil
+	}
+	triggerComment := "/approved"
+	approvers := a.GetArrayInput("approvers")
+	c := a.GithubClient(tc.Repo, tc.SCM.APIURL)
+	pr, err := c.MergedPullRequest(ctx, tc.Commit)
+	if err != nil {
+		return fmt.Errorf("failed to get the pull request: %w", err)
+	}
+	comments, err := c.getIssueComments(ctx, pr)
+	if err != nil {
+		return fmt.Errorf("failed to get the pull request comments: %w", err)
+	}
+	approved := slices.IndexFunc(comments, func(c githubIssueComment) bool {
+		if c.AuthorHasAccess() && c.HasCommand(triggerComment) {
+			return len(approvers) == 0 || slices.Contains(approvers, c.Author())
+		}
+		return false
+	})
+	if approved == -1 {
+		a.Errorf("Please comment %q on the pull request to approve the action: %s", triggerComment, pr.URL)
+		return errors.New("Approval checks failed")
+	}
```

Comment:
Adding a polling loop here to fetch comments, so it will hold the action until the user comments on the issue to approve this action,